src/Command/Argument/Search/CreateArguments.php

```diff
@@ -0,0 +1,656 @@
+<?php
+
+/*
+ * This file is part of the Predis package.
+ *
+ * (c) 2009-2020 Daniele Alessandri
+ * (c) 2021-2023 Till Krüss
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Predis\Command\Argument\Search;
+
+use InvalidArgumentException;
+use Predis\Command\Argument\ArrayableArgument;
+
+class CreateArguments implements ArrayableArgument
+{
+    /**
+     * @var array
+     */
+    private $arguments = [];
+
+    /**
+     * @var string[]
+     */
+    private $supportedDataTypesEnum = [
+        'hash' => 'HASH',
+        'json' => 'JSON',
+    ];
+
+    /**
+     * @var string[]
+     */
+    private $sortingEnum = [
+        'asc' => 'ASC',
+        'desc' => 'DESC',
+    ];
+
+    /**
+     * Specify data type for given index. To index JSON you must have the RedisJSON module to be installed.
+     *
+     * @param  string $modifier
+     * @return $this
+     */
+    public function on(string $modifier = 'hash'): self
```

Comment:
Why `modifier` is optional with a default value? If you want to use the default just don't call this function...

Reply:
It was like this before, then @chayim suggest to add it, but I agree we don't need default value if it's anyway will be applied

Reply:
Fixed